### PR TITLE
feat(backend): get_limit

### PR DESCRIPTION
**Pull request by Gift-Naomi**

### CODE REVIEW

File: backend/src/routes/getLimit.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    try {
+      // Use Drizzle ORM to find the balance by address
+      const currentLimit = await fastify.db.query.mockLimit
+        .findFirst({
+          where: eq(mockLimit.address, address),
+          columns: { limit: true },
+        })
+        .execute();
```
Comment (LucasLvy) [inline backend/src/routes/getLimit.ts:25]:
```suggestion
      const { currentLimit } = await fastify.db.select({limit: mockLimit.limit).from(mockLimit).where(eq(mockLimit.address, address));
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
you need to add tests + run `pnpm fix`

File: backend/src/routes/getLimit.ts (hunk 1 above)
Comment (Gift-Naomi) [reply backend/src/routes/getLimit.ts]: I have made the changes.

File: backend/test/getLimit.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    const response = await app.inject({
+      method: 'GET',
+      url: `/get_limit?address=${testAddress}`,
+    });
+
+    expect(response.statusCode).toBe(200);
+    expect(response.json()).toEqual({
+      limit: [
+        {
+          limit: '7714789860048896',
```
Comment (LucasLvy) [inline backend/test/getLimit.test.ts:48]:
```suggestion
          limit: '7714789860.048896',
```

File: backend/src/routes/getLimit.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if (!addressRegex.test(address)) {
+      return reply.status(400).send({ message: 'Invalid address format' });
+    }
+
+    try {
+      // Use Drizzle ORM to find the balance by address
+      const limit = await fastify.db
+        .select({ limit: mockLimit.limit })
+        .from(mockLimit)
+        .where(eq(mockLimit.address, address));
```
Comment (LucasLvy) [inline backend/src/routes/getLimit.ts:23]: limit shouldn't be an array it should be just the value

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: backend/src/db/schema.ts [hunk 4]
```diff
@@ -31,3 +31,8 @@ export const registration = pgTable('registration', {
   created_at: timestamp('created_at').defaultNow(),
   is_confirmed: boolean('is_confirmed').default(false),
 });
+
+export const mockLimit = pgTable('mock_limit', {
+  address: text('address').primaryKey(),
+  limit: text('limit'),
```
Comment (LucasLvy) [inline backend/src/db/schema.ts:37]:
```suggestion
  limit: text('limit'),
  blockTimestamp: timestamp('block_timestamp', { withTimezone: false }),
```

File: backend/src/routes/getLimit.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
```
Comment (LucasLvy) [inline backend/src/routes/getLimit.ts:23]:
```suggestion
      const limit = await fastify.db
        .select({ limit: mockLimit.limit })
        .from(mockLimit)
        .where(eq(mockLimit.address, address))
        .orderBy(desc(mockLimit.blockTimestamp))
        .limit(1);
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**